### ISSUE TITLE
Infinity-negative Infinity pricing

### ISSUE THREAD

**Issue by jeepgurl99** [mikebryant/ac-nh-turnip-prices#371]
**Things to check**
- Have you forced a refresh in your browser (e.g. ctrl+F5 in Chrome/Firefox)?
  - no (using in chrome from mobile)
- Have you made sure the first time buyer option is correct? If you're unsure, try it both ways
  - yes
- Have you time-travelled this week? Travelling backwards resets the prices, and therefore you must not put prices in from both before & after the time-travel
  - no

**Describe the bug**
When I entered my price today (641) which was higher than the guaranteed price predicted it changed to 999 to 0 with guaranteed Max and min of infinity.
**To Reproduce**

**Permalink to your prices**
Please click the `Copy Permalink` button and paste it here
https://turnipprophet.io?prices=102.93.88.85.80..190.641.....&pattern=3

**Expected behavior**
I wanted to update it so that it continued to keep gathering information.
**Screenshots**
![Screenshot_20200514-111935](https://user-images.githubusercontent.com/65355711/81952727-d8230300-95d4-11ea-8340-757728a11e83.png)
![Screenshot_20200514-111925](https://user-images.githubusercontent.com/65355711/81952732-d8230300-95d4-11ea-9390-215bbbf1d40d.png)
![Screenshot_20200514-111919](https://user-images.githubusercontent.com/65355711/81952735-d8bb9980-95d4-11ea-8572-d187d6f9e5b4.png)

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**
 - Device: Pixel 2
 - OS: no idea
 - Browser Chrome
 - Version no idea

**Additional context**

**Comment by Jesterw00t** [comment]
I ran into this as well, perhaps they've updated the algorithms recently?

Here is my prices for the week: https://turnipprophet.io?prices=106.95.91.99.165.597.198.84.....&pattern=3

This was tracked on my Windows 10 PC using Google Chrome.  
I've attached screenshots of my browser before and after putting in this morning's price which was outside the expected bounds.

![tunipsbroken](https://user-images.githubusercontent.com/6462024/81954158-84191e00-95d6-11ea-83c5-b57b84830ed1.png)
![turnipsprebroke](https://user-images.githubusercontent.com/6462024/81954168-87140e80-95d6-11ea-93d6-fcde3cf961f0.png)

**Comment by jdongg** [comment]
@Jesterw00t is it possible 84 was your PM price instead of AM? The large spike always has 
 2 stages of decrease after the peak before it bottoms out, so the jump from 198 to 84 doesn't seem possible.

@jeepgurl99 your prices themselves seem valid. If you remove the Daisy price, the calculator gives you no issues. Is it possible you just recorded rthe Daisy price incorrectly? The peak of the large spike pattern will always be between 2x and 6x your Daisy price. So Daisy @ 102 is inconsistent with a peak of 641 (the max possible is 612).

**Comment by Jesterw00t** [comment]
> @Jesterw00t is it possible 84 was your PM price instead of AM? The large spike always has
> 2 stages of decrease after the peak before it bottoms out, so the jump from 198 to 84 doesn't seem possible.
> 

Sorry for the delay in response.  I definitely tracked very closely.  In fact I'm back today with a whole new set that did not go as predicted.  

2 weeks ago I reset the game and started over for the fun of it.  This was the first week i could buy Turnips from Daisy Mae.  So I started tracking, as I put my numbers in over the course of the week the system was 100% positive I'd have a small spike either Thursday or Friday.  Instead I've had a steady decline.  

Here's the link to my numbers : 
https://turnipprophet.io?prices=108.97.92.88.84.80.77.71.66.62...&first=true

**Comment by theRTC204** [comment]
> 2 weeks ago I reset the game and started over for the fun of it. This was the first week i could buy Turnips from Daisy Mae. So I started tracking, as I put my numbers in over the course of the week the system was 100% positive I'd have a small spike either Thursday or Friday. Instead I've had a steady decline.

This is an interesting condition that the app is likely not able to handle (it's probably also too niche to warrant the work necessary to make it handle this).

What I mean is, the way the game was programmed, the very first time any resident of your island buys turnips on your island (visitors do not count), the game locks you into a Small Spike pattern.

```cpp
  if (checkGlobalFlag("FirstKabuBuy")) {
    if (!checkGlobalFlag("FirstKabuPattern")) {
      setGlobalFlag("FirstKabuPattern", true);
      whatPattern = 3;
    }
  }
```

The above code shows that if it is the first time buying, your pattern is locked to Pattern 3, which maps to Small Spike. You can also see that it stores the fact that it happened into a global flag.

What we do not (and perhaps cannot) know, is whether "resetting the game" would also reset this flag. It is very possible that this flag is stored in memory on the Switch in a place that is not directly effected by your game save. It could be that it requires the entire Switch to be factory reset/wiped for this flag to be unset. It's also possible that simply resetting the game save would accomplish this.

What I would say is, this week is clearly an anomoly, and for that reason, it's not really worth spending the energy necessary to investigate and implement a fix for, because the behavior you describe (resetting the game save) is not likely something that majority of users would ever do, and even those who do it will only be impacted by the error once in their entire playthough.

**Comment by Jesterw00t** [comment]
Would it possibly store that flag somewhere other than the game's save data?  When I reset, I wiped the save data from the switch.  So as far as the game new it was the first time it'd ever been booted.

**Comment by theRTC204** [comment]
Sorry, I believe I may have confused you. The snippet of code I showed is from the video game itself; it is not code in this application. We have no affiliation with the game, it's developers, or Nintendo in any way.

**Comment by Jesterw00t** [comment]
Oh I understand that you don't have affiliation with the game, I just
thought you mean the game's algorithm.  I did a reset on the turnip.prophet
app before starting the week out too.

**Comment by theRTC204** [comment]
What I meant in my original comment was that the global flag is part of the game code, not part of this web app's code. We have no control over where that value is stored, so your question about whether the value can be stored somewhere else is not something we can influence at all.

**Comment by Jesterw00t** [comment]
Not sure what's going on but its happened to me again.

https://turnipprophet.io?prices=91.73.68.64.103.154.161......  This week's numbers.

**Comment by cutewhensedated** [comment]
This also happened to me - with a minimum of "infinity" and a maximum of "-infinity". I haven't time traveled, I'm using a MacBook with Google Chrome. I put everything in accurately and I know it was accurate because I kept a log of it via messages among a group of friends in a private Discord. 
It was fine before I hit the 367 bells/turnip amount yesterday morning. And by that I mean that it looked like my friends' turnip prophet graphs do, without this whole impossible min of infinity and max of negative infinity. That part makes no sense whatsoever.
[https://turnipprophet.io?prices=107.93.112.128.214.367.214.147.75....&pattern=3](My outcomes this week.)
<img width="1246" alt="Screen Shot 2020-09-03 at 12 12 25 PM" src="https://user-images.githubusercontent.com/70721747/92158822-5522f400-ede1-11ea-8e36-44266f76139f.png">
<img width="1247" alt="Screen Shot 2020-09-03 at 12 12 57 PM" src="https://user-images.githubusercontent.com/70721747/92158838-5ce29880-ede1-11ea-8806-4fd2720d3d90.png">
<img width="1247" alt="Screen Shot 2020-09-03 at 12 13 23 PM" src="https://user-images.githubusercontent.com/70721747/92158858-61a74c80-ede1-11ea-893c-6a2f5e00c535.png">
Let me know if there's any additional information you need. I'm only giving you guys this so you have the information, I'm not worried about the inaccuracy for my own sake. But, by all means, if you want me to change something to see what happens or have any other questions, I have no problem doing so to help.
